File: phrase_reversals.rb
# http://rosettacode.org/wiki/Phrase_reversals

test_string = "rosetta code phrase reversal"

def phrase_reversals(str)
  # reverse the string
  puts str.reverse
  # reverse each individual word in the string, keeping order
  str.split(" ").map.each do |word|
    print "#{word.reverse} " 
  end; puts
  # reverse the words in the string without reversing the words
  str.split(" ").to_a.reverse.each do |word|
    print "#{word} "
  end; puts  
  nil
end

puts phrase_reversals(test_string)    

File: phrase_reversals.swift
#!/usr/bin/env swift
// http://rosettacode.org/wiki/Phrase_reversals

import Foundation

func phraseReversal(s: String) -> (String, String, String) {
  let a = String(s.characters.reverse())
  let b = s.characters.split { $0 == " " }.map(String.init).map { String($0.characters.reverse()) }.joinWithSeparator(" ")
  let c = s.characters.split { $0 == " " }.map(String.init).reverse().joinWithSeparator(" ")
  return (a, b, c)
}

let t = "rosetta code phrase reversal"
print(phraseReversal(t))
